openreview/venue/venue.py

```diff
@@ -1073,50 +1073,49 @@ def ithenticate_create_and_upload_submission(self):
-                group_id=self.get_submission_id(),
-                group_context={
-                    "id": self.id,
-                    "name": self.name,
-                    "owners": [
-                        # {
-                        #     "id": "d7cf2650-c1c7-11e8-b568-0800200c9a66",
-                        #     "family_name": "test_instructor_first_name",
-                        #     "given_name": "test_instructor_last_name",
-                        #     "email": "instructor_email@test.com"
-                        # },
-                        # {
-                        #     "id": "7a62f070-c265-11e8-b568-0800200c9a66",
-                        #     "family_name": "test_instructor_2_first_name",
-                        #     "given_name": "test_instrutor_2_last_name",
-                        #     "email": "intructor_2_email@test.com"
-                        # }
-                    ],
-                },
-                group_type="ASSIGNMENT",
-            )
-            iThenticate_submission_id = res["id"]
+            if (submission.id not in edges_dict):
+                name = openreview.tools.pretty_id(submission.signatures[0])
+
+                res = iThenticate_client.create_submission(
+                    owner=submission.signatures[0],
+                    title=submission.content["title"]["value"],
+                    timestamp=datetime.datetime.fromtimestamp(
+                        submission.tcdate / 1000, tz=datetime.timezone.utc
+                    ).strftime("%Y-%m-%dT%H:%M:%SZ"),
+                    owner_first_name=name.split(" ", 1)[0],
+                    owner_last_name=name.split(" ", 1)[1],
+                    owner_email=self.client.get_profile(
+                        submission.signatures[0]
+                    ).get_preferred_email(),
+                    group_id=self.get_submission_id(),
+                    group_context={
+                        "id": self.id,
+                        "name": self.name,
+                        "owners": [
+                            # {
+                            #     "id": "d7cf2650-c1c7-11e8-b568-0800200c9a66",
+                            #     "family_name": "test_instructor_first_name",
+                            #     "given_name": "test_instructor_last_name",
+                            #     "email": "instructor_email@test.com"
+                            # },
+                            # {
+                            #     "id": "7a62f070-c265-11e8-b568-0800200c9a66",
+                            #     "family_name": "test_instructor_2_first_name",
+                            #     "given_name": "test_instrutor_2_last_name",
+                            #     "email": "intructor_2_email@test.com"
+                            # }
+                        ],
+                    },
+                    group_type="ASSIGNMENT",
+                )
+                iThenticate_submission_id = res["id"]
```

Comment:
I would suggest to add an else and print the submission that already have an edge so it can help us to know what is happening.

Reply:
Done